packages/react-openapi/src/OpenAPISchema.tsx

```diff
@@ -21,84 +20,60 @@ export interface OpenAPISchemaPropertyEntry {
-    const properties = parentCircularRef ? null : getSchemaProperties(schema);
-    const alternatives = parentCircularRef
-        ? null
-        : getSchemaAlternatives(schema, new Set(circularRefs.keys()));
+    const { schema } = property;
 
-    if (alternatives?.[0]?.length) {
-        return (
-            <OpenAPISchemaAlternativesItem
-                {...props}
-                circularRefs={circularRefs}
-                context={context}
-                alternatives={alternatives}
-                parentCircularRef={parentCircularRef}
-            />
-        );
-    }
-
-    if ((properties && properties.length > 0) || schema.type === 'object') {
-        return (
-            <InteractiveSection id={id} className={clsx('openapi-schema', className)}>
-                <OpenAPISchemaPresentation {...props} />
-                {properties && properties.length > 0 ? (
-                    <OpenAPIDisclosure context={context} label={getDisclosureLabel(schema)}>
-                        <OpenAPISchemaProperties
-                            properties={properties}
-                            circularRefs={circularRefs}
-                            context={context}
-                        />
-                    </OpenAPIDisclosure>
-                ) : null}
-                {parentCircularRef ? (
-                    <OpenAPISchemaCircularRef id={parentCircularRef} schema={schema} />
-                ) : null}
-            </InteractiveSection>
-        );
-    }
+    const id = useId();
 
     return (
-        <InteractiveSection id={id} className={clsx('openapi-schema', className)}>
-            <OpenAPISchemaPresentation {...props} />
-            {(properties && properties.length > 0) ||
-            (schema.enum && schema.enum.length > 0) ||
-            parentCircularRef ? (
-                <>
-                    {properties?.length ? (
-                        <OpenAPISchemaProperties
-                            properties={properties}
-                            circularRefs={circularRefs}
-                            context={context}
-                        />
-                    ) : null}
-                    {parentCircularRef ? (
-                        <OpenAPISchemaCircularRef id={parentCircularRef} schema={schema} />
-                    ) : null}
-                </>
-            ) : null}
-        </InteractiveSection>
+        <div id={id} className={clsx('openapi-schema', className)}>
+            <OpenAPISchemaPresentation property={property} />
+            {(() => {
+                const parentCircularRef = parentCircularRefs.get(schema);
+
+                // Avoid recursing infinitely, and instead render a link to the parent schema
+                if (parentCircularRef) {
+                    return <OpenAPISchemaCircularRef id={parentCircularRef} schema={schema} />;
+                }
+
+                const circularRefs = parentCircularRefs.set(schema, id);
```

Comment:
I am mutating the parentCircularRefs which is bad but I don't understand why it does not work else...

I suspect streaming and Suspense to cut the rendering, start it again and never reach a point where it works.